{{cookiecutter.repo_name}}/.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,167 @@
+name: GH Actions CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
+    # Weekly tests run on main by default:
+    #   Scheduled workflows run on the latest commit on the default or base branch.
+    #   (from https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule)
+    - cron: "0 0 * * 0"
+
+concurrency:
+  # Specific group naming so CI is only cancelled
+  # within same PR or on merge to main
+  group: {{ "${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}" }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  main_tests:
+    if: "github.repository == '{{ cookiecutter.github_username }}/{{ cookiecutter.repo_name }}'"
+    runs-on: {{ '${{ matrix.os }}' }}
+    strategy:
+        fail-fast: false
+        matrix:
+          os: [macOS-latest, ubuntu-latest, windows-latest]
+          python-version: ["3.8", "3.9", "3.10"]
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Build information
+      run: |
+        uname -a
+        df -h
+        ulimit -a
+
+{% if cookiecutter.dependency_source == 'Dependencies from pip only (no conda)' %}
+    - name: Set up Python {{ '${{ matrix.python-version }}' }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: {{ '${{ matrix.python-version }}' }}
+
+    - name: Install testing dependencies
+      # need non-login bash shell for Mac outside conda
+      # otherwise it sources system Python
+      shell: bash
```

Comment:
That's an annoying bug, could we just wrap around the defaults instead to make things cleaner?

Reply:
Smart! Done